src/AnonymousUser/AnonymousUserMiddleware.cs

```diff
@@ -52,16 +56,22 @@ private async Task HandleRequestAsync(HttpContext httpContext)
                 uid = _options.UserIdentifierFactory.Invoke(httpContext);
                 var encodedUid = await cookieEncoder.EncodeAsync(uid);
 
-                var cookieOptions = new CookieOptions
-                {
-                    Expires = _options.Expires,
-                };
+                var cookieOptions = new CookieOptions {Expires = _options.Expires,};
 
                 httpContext.Response.Cookies.Append(_options.CookieName, encodedUid, cookieOptions);
             }
 
-            var identity = new ClaimsIdentity(new[] { new Claim(_options.ClaimType, uid) });
-            httpContext.User.AddIdentity(identity);
+            var claim = new Claim(_options.ClaimType, uid);
+
+            if (httpContext.User.Identity is ClaimsIdentity ci)
+            {
+                ci.AddClaim(claim);
+            }
+            else
+            {
+                var identity = new ClaimsIdentity(new[] { claim });
+                httpContext.User.AddIdentity(identity);
+            }
```

Comment:
If `User.Identity` is not a `ClaimsIdentity` you would just override it. Is that ok?

Reply:
In this case it would not override it, but add another `ClaimsIdentity` to the list of identities. That should be fine as most identities are based off `ClaimsIdentity` and we don't support custom `IIdentity` implementations.